proxy/provider.go

```diff
@@ -206,14 +206,14 @@ Returns:
 	proxy: A proxy is found through environment variables for the given traffic protocol.
 	nil: No proxy is found or an error occurs reading the environment variables.
 */
-func (p *provider) readSystemEnvProxy(protocol string, targetUrl *url.URL) Proxy {
+func (p *provider) readSystemEnvProxy(prefix string, targetUrl *url.URL) Proxy {
 	// SOCKS configuration is set as ALL_PROXY and all_proxy on Linux. Replace here for all OSs to keep consistent
-	if protocol == "socks" {
-		protocol = "all"
+	if prefix == "socks" {
```

Comment:
It might make sense to do a "HasPrefix" call here (see the link in the previous comment where the scheme could be socks4:// or socks5://).

Reply:
:+1: